src/main/java/org/example/controller/UserController.java

```diff
@@ -0,0 +1,60 @@
+package org.example.controller;
+
+import org.example.model.ERole;
+import org.example.model.User;
+import org.example.service.AccountHistoryService;
+import org.springframework.web.bind.annotation.*;
+import org.example.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private final UserService userService;
+    private final AccountHistoryService accountHistoryService;
+
+    @Autowired
+    public UserController(UserService userService, AccountHistoryService accountHistoryService) {
+        this.userService = userService;
+        this.accountHistoryService = accountHistoryService;
+    }
+
+    @PostMapping(value = "/register")
+    public ResponseEntity<User> saveUser(@RequestBody User user) {
+        User savedUser = this.userService.saveUser(user);
+
+        if(savedUser == null){
+            return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+        } else{
+            return new ResponseEntity<>(savedUser, HttpStatus.OK);
+        }
+    }
+
+    @GetMapping(value = "/getInfo")
+    public ResponseEntity<User> getUser(@RequestParam String email){
```

Comment:
this is not restful

Reply:
What would be restful?

Reply:
https://www.sitepoint.com/rest-api/ see here

Reply:
for a get you ll have /user in path not /getInfo